static/js/rest.js

```diff
@@ -12,8 +13,9 @@ import {
 
 export type Endpoint = {
   name: string,
-  url: string,
-  makeOptions: (...args: any) => Object
+  url?: string,
+  urlTemplate?: (params: ?Object) => string,
```

Comment:
in micromasters I handled this issue (sometimes we have a static url, other times it has a url parameter like `/api/v0/resource/${id}`) by just doing a check in the `makeFetchFunc` function like:

```js
fetchJSONWithCSRF(_.isFunction(url) ? url(...args) : url);
```

where `url` is the `url` property of the`Endpoint` object and `args` is the rest parameter to the function returned by `makeFetchFunc`. basically if we have a dynamic url we set the `url` property to be a function which expects the same parameters as the function for generating the request body and returns a string, and if the url is not dynamic we set the `url` property to just a string. Sometimes polymorphism is fun! haha. Anyway, either approach works fine I think.